src/path.jl

```diff
@@ -767,41 +767,37 @@ Base.write(fp::AbstractPath, x) = open(io -> write(io, x), fp, "w")
 # Default `touch` will just write an empty string to a file
 Base.touch(fp::AbstractPath) = write(fp, "")
 
-Base.tempname(::Type{<:AbstractPath}) = Path(tempname())
-tmpname() = tempname(SystemPath)
-
-Base.tempdir(::Type{<:AbstractPath}) = Path(tempdir())
-tmpdir() = tempdir(SystemPath)
-
-Base.mktemp(P::Type{<:AbstractPath}) = mktemp(tempdir(P))
-mktmp() = mktemp(SystemPath)
-
-Base.mktemp(fn::Function, P::Type{<:AbstractPath}) = mktemp(fn, tempdir(P))
-mktmp(fn::Function) = mktemp(fn, SystemPath)
-
-Base.mktempdir(P::Type{<:AbstractPath}) = mktempdir(tempdir(P))
-mktmpdir() = mktempdir(SystemPath)
-
-Base.mktempdir(fn::Function, P::Type{<:AbstractPath}) = mktempdir(fn, tempdir(P))
-mktmpdir(fn::Function) = mktempdir(fn, SystemPath)
+Base.tempname(P::Type{<:AbstractPath}; kwargs...) = tempname(tempdir(P); kwargs...)
+Base.mktemp(P::Type{<:AbstractPath}; kwargs...) = mktemp(tempdir(P); kwargs...)
+Base.mktemp(fn::Function, P::Type{<:AbstractPath}; kwargs...) = mktemp(fn, tempdir(P); kwargs...)
+Base.mktempdir(P::Type{<:AbstractPath}; kwargs...) = mktempdir(tempdir(P); kwargs...)
+Base.mktempdir(fn::Function, P::Type{<:AbstractPath}; kwargs...) = mktempdir(fn, tempdir(P); kwargs...)
+
+function Base.tempname(parent::AbstractPath; prefix="jl_", cleanup=true)
+    isdir(parent) || throw(ArgumentError("$(repr(parent)) is not a directory"))
+    fp = parent / string(prefix, uuid1())
```

Comment:
Switching to `uuid1` over `uuid4`. For older releases of Julia, these function depended on the `GLOBAL_RNG` and could in theory result in collisions if folks were resetting while generating temp files and directories.